File: Examples/iOS/SongLibraryPlayer/SongLibraryPlayer/AlbumsViewController.m
//
//  AlbumsViewController.m
//  SongLibraryPlayer
//
//  Created by Aurelius Prochazka on 12/26/13.
//  Copyright (c) 2013 Aurelius Prochazka. All rights reserved.
//

#import "AlbumsViewController.h"
#import "SongsViewController.h"
#import <AVFoundation/AVFoundation.h>
#import <MediaPlayer/MediaPlayer.h>

@interface AlbumsViewController ()
@property (nonatomic, strong) NSMutableArray *albumsList;
@end

@implementation AlbumsViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

    // Uncomment the following line to preserve selection between presentations.
    // self.clearsSelectionOnViewWillAppear = NO;

    
    MPMediaPropertyPredicate *artistPredicate =
    [MPMediaPropertyPredicate predicateWithValue:self.artistName
                                     forProperty:MPMediaItemPropertyArtist
                                  comparisonType:MPMediaPredicateComparisonContains];
    
    MPMediaQuery *albumsQuery = [MPMediaQuery albumsQuery];
    [albumsQuery addFilterPredicate:artistPredicate];
    
    self.albumsList = [NSMutableArray arrayWithArray:[albumsQuery collections]];
    [self.tableView reloadData];
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    // Return the number of sections.
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    // Return the number of rows in the section.
    return self.albumsList.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *CellIdentifier = @"AlbumCell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier];
    NSString *albumTitle = [[[self.albumsList objectAtIndex:indexPath.row] representativeItem] valueForProperty:MPMediaItemPropertyAlbumTitle];
    cell.textLabel.text = albumTitle;
    return cell;
}

#pragma mark - Navigation

// In a story board-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    UITableViewCell *senderCell = (UITableViewCell *)sender;
    SongsViewController *vc = [segue destinationViewController];
    vc.albumName = senderCell.textLabel.text;
    vc.title = senderCell.textLabel.text;
}


@end

File: Examples/iOS/SongProcessor/SongProcessor/View Controllers/iTunes Library Access/AlbumsViewController.swift
//
//  AlbumsViewController.swift
//  SongProcessor
//
//  Created by Aurelius Prochazka on 6/22/16.
//  Copyright © 2016 AudioKit. All rights reserved.
//

import AVFoundation
import MediaPlayer
import UIKit

class AlbumsViewController: UITableViewController {

    var artistName: String!
    var albumsList = [MPMediaItemCollection]()

    override func viewDidLoad() {
        super.viewDidLoad()

        let artistPredicate = MPMediaPropertyPredicate(value: artistName,
                                                       forProperty: MPMediaItemPropertyArtist,
                                                       comparisonType: .contains)

        let albumsQuery = MPMediaQuery.albums()
        albumsQuery.addFilterPredicate(artistPredicate)

        if let list = albumsQuery.collections {
            albumsList = list
            tableView.reloadData()
        }
    }

    // MARK: - Table view data source

    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return albumsList.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {

        let cellIdentifier = "AlbumCell"
        let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier) ??
            UITableViewCell(style: .default, reuseIdentifier: cellIdentifier)

        // Configure the cell...
        if let repItem = albumsList[(indexPath as NSIndexPath).row].representativeItem,
            let albumTitle = repItem.value(forProperty: MPMediaItemPropertyAlbumTitle) as? String {
            cell.textLabel?.text = albumTitle
        }

        return cell
    }

    // MARK: - Navigation

    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {

        if segue.identifier == "SongsSegue" {
            if let senderCell = sender as? UITableViewCell, let songsVC = segue.destination as? SongsViewController {
                songsVC.artistName = artistName
                songsVC.albumName = senderCell.textLabel?.text
                songsVC.title = senderCell.textLabel?.text
            }
        }

    }

}
